prow/kube/metrics.go

```diff
@@ -113,11 +119,21 @@ var previousStates map[jobIdentifier]prowapi.ProwJobState
 
 // GatherProwJobMetrics gathers prometheus metrics for prowjobs.
 // Not threadsafe, ensure this is called serially.
-func GatherProwJobMetrics(current []prowapi.ProwJob) {
+func GatherProwJobMetrics(l *logrus.Entry, current []prowapi.ProwJob) {
```

Comment:
You can just do logrus.WithError(err).Debug()" if you don't want to pass in an entry.

Reply:
That's good to know. In this case I do want log with more context, so that this debug log can tell which component it's calling from.